waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -195,13 +198,31 @@ proc selectPeer*(pm: PeerManager, proto: string): Option[PeerInfo] =
   else:
     return none(PeerInfo)
 
-proc reconnectPeers*(pm: PeerManager, proto: string) {.async.} =
+proc reconnectPeers*(pm: PeerManager, proto: string, backoff: chronos.Duration = chronos.seconds(0)) {.async.} =
   ## Reconnect to peers registered for this protocol. This will update connectedness.
   ## Especially useful to resume connections from persistent storage after a restart.
   
   debug "Reconnecting peers", proto=proto
   
   for storedInfo in pm.peers(proto):
+    # Check if peer is reachable.
+    if pm.peerStore.connectionBook.get(storedInfo.peerId) == CannotConnect:
+      debug "Not reconnecting to unreachable peer", peerId=storedInfo.peerId
+      continue
+    
+    # Respect optional backoff period where applicable.
+    let
+      disconnectTime = Moment.init(pm.peerStore.disconnectBook.get(storedInfo.peerId), Second)  # Convert 
```

Comment:
Curious why this conversion is needed?

Reply:
`chronos` library makes use of the concepts [`Moment` and `Duration`](https://github.com/status-im/nim-chronos/blob/master/chronos/timer.nim#L180-L182), which on their own has no direct meaning but is a fast and efficient way to compare different moments in time.  The `Duration` used to `sleepAsync`, for example, points to this library.
In contrast, the Nim `times` module defines absolute time and duration in terms of the epoch/system time (`getTime()`). This is a roundabout way of saying, that since I need to compare Moments across restarts I convert a saved epoch time from `std/times` to a Moment in `chronos` for ease of use. Hope this explanation makes sense. 😁

Reply:
Yeah, that makes sense! thanks!